### DISCUSSION TITLE
Angular filter for DataList so that "name" or "description" can be used in name templates?

### DISCUSSION THREAD

**Discussion by hfloyd** [leekelleher/umbraco-contentment#398] [Question] [answered]
I love using DataList, and now that I'm using more and more "Block Lists" in sites, I often have a desire to use a DataList value in the name template, with an inline block editor. Using the property alias will return the value, but usually, the "name" would be more friendly. I know that Angular filters can be used to grab the name of a Content, Media or Member node (ex: `{{ myPropertyAlias | ncNodeName }}`), so I wonder if an Angular filter could be created to retrieve a different property from the DataList data source?

Has anyone attempted this?

**Comment by leekelleher** [comment, marked as answer]
Hey @hfloyd!

Given the nature of Data List, where the data-source items could come from anywhere, it would add complications to developing a built-in solution to support AngularJS filters.

Thinking outside the box, there are a couple of ways that it could be done, but you'd need to know some CMS data, like the data-type's Guid. If it's for a small-ish dataset, (e.g. not hundreds/thousands), then you could hook into the `ServerVariablesParsingNotification` event, fetch the data-source items from the data-type. (I have [some code that does this](https://github.com/leekelleher/umbraco-contentment/blob/4.7.0/src/Umbraco.Community.Contentment/DataEditors/DataList/DataListApiController.cs#L106), you'll probably need to copy it, as it's in an API controller).

Then your AngularJS filter could look something like this...

	angular.module("umbraco.filters").filter("myDataListItems", [
		"$filter",
		function ($filter) {
			return function (value) {
				var lookup = Umbraco.Sys.ServerVariables.myDataListItems;
				return Object.keys(lookup).find(key => lookup[key] === value);
			};
		}
	]);

The "`myDataListItems`" bit is whatever you want to call it in the `ServerVariablesParsingNotification` code.

I think that'd be the most performant way of handling it, otherwise the way `ncNodeName` does it is to make a server request per each node ID, then cache it in browser memory - which is fine, but that initial population can be very chatty (with server requests).

Hope this helps, or at least helps to spark an idea.

**Reply by hfloyd** [reply]
Thanks, @leekelleher When I have some time I'll take a look at your code sample... this is generally for smaller (manually-provided) lists of items.
